CI Build Failure: `wasm32v1-none` Target Not Installed for Soroban Contracts

## Description

The Soroban smart contracts CI pipeline is failing during contract builds due to a missing Rust target required for Soroban contract compilation.

The CI logs repeatedly show failures while building multiple contracts using both `stellar contract build` and fallback `cargo rustc`.

---

## Error Logs

```text
error[E0463]: can't find crate for `core`
= note: the `wasm32v1-none` target may not be installed
= help: consider downloading the target with `rustup target add wasm32v1-none`
```


## Observed Behavior

CI fails during the build step for Soroban smart contracts

Multiple contracts fail with the same Rust core crate error

The pipeline attempts to enter a non-existent contract directory

CI exits with a non-zero status, blocking pull requests

## Expected Behavior

All existing Soroban contracts should build successfully in CI

The CI pipeline should only attempt to build valid contract directories

Build failures should be deterministic and clearly reported

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -67,6 +67,7 @@ jobs:
           rustc --version
           cargo --version
           rustup target list --installed | grep wasm32-unknown-unknown || rustup target add wasm32-unknown-unknown
+          rustup target list --installed | grep wasm32v1-none || rustup target add wasm32v1-none || echo "Warning: wasm32v1-none target not available, will use wasm32-unknown-unknown fallback"
 
       - name: Build workspace (WASM)
         run: |
@@ -75,7 +76,7 @@ jobs:
 
       - name: Build Soroban contracts
         run: |
-          for contract in remittance_split savings_goals bill_payments insurance family_wallet; do
+          for contract in remittance_split savings_goals bill_payments insurance; do
             echo "Building contract: $contract"
             cd $contract
             if command -v soroban &> /dev/null; then
@@ -93,7 +94,7 @@ jobs:
 
       - name: Verify WASM files exist
         run: |
-          for contract in remittance_split savings_goals bill_payments insurance family_wallet; do
+          for contract in remittance_split savings_goals bill_payments insurance; do
             wasm_file="target/wasm32-unknown-unknown/release/${contract}.wasm"
             if [ ! -f "$wasm_file" ]; then
               echo "Error: WASM file not found: $wasm_file"

